src/main/drivers/vtx_smartaudio.c

```diff
@@ -0,0 +1,780 @@
+       adjdir = -1;
+       dprintf(("autobaud: now going down\r\n"));
+    } else if ((adjdir == -1 && smartbaud == SMARTBAUD_MIN)) {
+       adjdir = 1;
+       dprintf(("autobaud: now going up\r\n"));
+    }
+
+    smartbaud += baudstep * adjdir;
+
+    dprintf(("autobaud: %d\r\n", smartbaud));
+
+    smartAudioSerialPort->vTable->serialSetBaudRate(smartAudioSerialPort, smartbaud);
+
+    sa_pktsent = 0;
+    sa_pktrcvd = 0;
+}
+
+// Transport level protocol variables
+
+static uint32_t sa_lastTransmission = 0;
+static uint8_t sa_outstanding = SA_CMD_NONE; // Outstanding command
+static uint8_t sa_osbuf[32]; // Outstanding comamnd frame for retransmission
+static int sa_oslen;         // And associate length
+
+static void saProcessResponse(uint8_t *buf, int len)
+{
+    uint8_t resp = buf[0];
+
+    if (resp == sa_outstanding) {
+        sa_outstanding = SA_CMD_NONE;
+    } else if ((resp == SA_CMD_GET_SETTINGS_V2) && (sa_outstanding == SA_CMD_GET_SETTINGS)) {
+        sa_outstanding = SA_CMD_NONE;
+    } else {
+        saerr_oooresp++;
+        dprintf(("processResponse: outstanding %d got %d\r\n", outstanding, resp));
+    }
+
+    switch(resp) {
+    case SA_CMD_GET_SETTINGS_V2: // Version 2 Get Settings
+    case SA_CMD_GET_SETTINGS: // Version 1 Get Settings
+        if (len < 7)
+            break;
+
+        sa_vers = (buf[0] == SA_CMD_GET_SETTINGS) ? 1 : 2;
+        sa_chan = buf[2];
+        sa_power = buf[3];
+        sa_opmode = buf[4];
+        sa_freq = (buf[5] << 8)|buf[6];
+
+        smartAudioPrintSettings();
+
+#ifdef SMARTAUDIO_DEBUG_MONITOR
+        debug[0] = sa_vers * 100 + sa_opmode;
+        debug[1] = sa_chan;
+        debug[2] = sa_freq;
+        debug[3] = sa_power;
+#endif
+
+        break;
+
+    case SA_CMD_SET_POWER: // Set Power
+        break;
+
+    case SA_CMD_SET_CHAN: // Set Channel
+        break;
+
+    case SA_CMD_SET_FREQ: // Set Frequency
+        if (len < 5)
+            break;
+
+        if (sa_pitfreqpending)
+            sa_pitfreq = ((buf[2] << 8)|buf[3]) & ~SA_FREQ_GETPIT;
+        break;
+
+    case SA_CMD_SET_MODE: // Set Mode
+        dprintf(("resp SET_MODE 0x%x\r\n", buf[2]));
+        break;
+    }
+}
+
+static void saReceiveFramer(uint8_t c)
+{
+
+    static enum saFramerState_e {
+        S_WAITPRE1, // Waiting for preamble 1 (0xAA)
+        S_WAITPRE2, // Waiting for preamble 2 (0x55)
+        S_WAITRESP, // Waiting for response code
+        S_WAITLEN,  // Waiting for length
+        S_DATA,     // Receiving data
+        S_WAITCRC,  // Waiting for CRC
+    } state = S_WAITPRE1;
+
+    static int len;
+    static int dlen;
+
+    switch(state) {
+    case S_WAITPRE1:
+        if (c == 0xAA) {
+            state = S_WAITPRE2;
+        } else {
+            state = S_WAITPRE1; // Bogus
+        }
+        break;
+
+    case S_WAITPRE2:
+        if (c == 0x55) {
+            state = S_WAITRESP;
+        } else {
+            state = S_WAITPRE1;
```

Comment:
What would happen if I read `0xaa 0xaa 0x55 0x09 0x06 0x01 0x00 0x1a 0x16 0xe9 0xda` (a valid v2 get settings reply with an extra `0xaa` in front) from the VTX? If I understood the implementation right, I think that the second byte would put you back into the `PRE1` state but also discard the byte that would make the rest of the message a valid packet. This same issue exists for the subsequent branches that bring us back into `PRE1`.

Would it be worthwhile to handle this more gracefully so that messages with useful information, such as the above example, could be read rather than having to be discarded entirely?